From c944a8eacc11895c42e7ff5ee61fe70c1f3d32be Mon Sep 17 00:00:00 2001
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 31 Dec 2021 15:59:09 +0800
Subject: [PATCH] kabi: add KABI_SIZE_ALIGN_CHECKS for more stringent kabi
 checks
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: c944a8eacc11895c42e7ff5ee61fe70c1f3d32be


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4K3S5

This option enables more stringent kabi checks. Those must be
disable in case of a debug-build because they allow to change struct
sizes.

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 Kconfig | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Kconfig b/Kconfig
index 745bc773f567..bb409bb587b0 100644
--- a/Kconfig
+++ b/Kconfig
@@ -30,3 +30,10 @@ source "lib/Kconfig"
 source "lib/Kconfig.debug"
 
 source "Documentation/Kconfig"
+
+config KABI_SIZE_ALIGN_CHECKS
+	bool "Enables more stringent kabi checks in the macros"
+	default y
+	help
+	  This option enables more stringent kabi checks. Those must be disable
+	  in case of a debug-build because they allow to change struct sizes.
-- 
2.26.2

